src/libstat/stat_process.c

```diff
@@ -1262,3 +1263,33 @@ rspamd_stat_statistics(struct rspamd_task *task,
 
 	return RSPAMD_STAT_PROCESS_OK;
 }
+
+rspamd_stat_result_t
+rspamd_stat_autolearn(struct rspamd_task *task, unsigned int stage,
+					 GError **err)
+{
+	struct rspamd_stat_ctx *st_ctx;
+	rspamd_stat_result_t ret = RSPAMD_STAT_PROCESS_OK;
+
+	st_ctx = rspamd_stat_get_ctx();
+	g_assert(st_ctx != NULL);
+
+	if (st_ctx->classifiers->len == 0) {
+		task->processed_stages |= stage;
+		return ret;
+	}
+
+	if (stage == RSPAMD_TASK_STAGE_AUTOLEARN_PRE) {
+		rspamd_stat_cache_check(st_ctx, task, task->classifier, task->flags & RSPAMD_TASK_FLAG_LEARN_SPAM, &err);
```

Comment:
In this case, we will check stat cache twice: once on autolearn and one on the actual learning.